tasks/20230529-avalanche-root-gauge-factory/test/task.fork.ts

```diff
@@ -0,0 +1,374 @@
+    gaugeAdder = await gaugeAdderTask.deployedInstance('GaugeAdder');
+
+    const balancerTokenAdminTask = new Task('20220325-balancer-token-admin', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    BALTokenAdmin = await balancerTokenAdminTask.deployedInstance('BalancerTokenAdmin');
+
+    BAL = await BALTokenAdmin.getBalancerToken();
+
+    const gaugeControllerTask = new Task('20220325-gauge-controller', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    gaugeController = await gaugeControllerTask.deployedInstance('GaugeController');
+    veBAL = await gaugeControllerTask.instanceAt(
+      'VotingEscrow',
+      gaugeControllerTask.output({ network: 'mainnet' }).VotingEscrow
+    );
+
+    const weightedPoolTask = new Task('20210418-weighted-pool', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    bal80weth20Pool = await weightedPoolTask.instanceAt('WeightedPool2Tokens', VEBAL_POOL);
+  });
+
+  describe('Factory bridge limits', () => {
+    const randomInt = (max: number) => Math.floor(Math.random() * Math.floor(max));
+
+    it('stores the original bridge limits', async () => {
+      const { minBridgeAmount, maxBridgeAmount } = await factory.getAvalancheBridgeLimits();
+
+      expect(minBridgeAmount).to.eq(MIN_BRIDGE_LIMIT);
+      expect(maxBridgeAmount).to.eq(MAX_BRIDGE_LIMIT);
+    });
+
+    describe('setting bridge limits', async () => {
+      const factor = randomInt(10) + 1;
+
+      const newLowerLimit = MIN_BRIDGE_LIMIT.mul(factor);
+      const newUpperLimit = MAX_BRIDGE_LIMIT.mul(factor);
+
+      context('without permission', () => {
+        it('fails if the caller has no permission', async () => {
+          await expect(factory.setAvalancheBridgeLimits(newLowerLimit, newUpperLimit)).to.be.revertedWith('BAL#401');
+        });
+      });
+
+      context('with permission', () => {
+        before('grant permission', async () => {
+          const setLimitsAction = await actionId(factory, 'setAvalancheBridgeLimits');
+
+          await authorizer.connect(daoMultisig).grantRole(setLimitsAction, admin.address);
+        });
+
+        it('allows updating bridge limits', async () => {
+          const tx = await factory.connect(admin).setAvalancheBridgeLimits(newLowerLimit, newUpperLimit);
+          expectEvent.inReceipt(await tx.wait(), 'AvalancheBridgeLimitsModified', {
+            minBridgeAmount: newLowerLimit,
+            maxBridgeAmount: newUpperLimit,
+          });
+
+          const { minBridgeAmount, maxBridgeAmount } = await factory.getAvalancheBridgeLimits();
+
+          expect(minBridgeAmount).to.eq(newLowerLimit);
+          expect(maxBridgeAmount).to.eq(newUpperLimit);
+        });
+
+        it('ensure valid bridge limits', async () => {
+          // Reverse the limits; max must be > min
+          await expect(
+            factory.connect(admin).setAvalancheBridgeLimits(newUpperLimit, newLowerLimit)
+          ).to.be.revertedWith('Invalid Bridge Limits');
+        });
+      });
+    });
+  });
```

Comment:
I was trying to limit the unit test to only stuff that was hard to test in a fork, but sure, I could move it. Normally we don't have unit tests at all, but since we do, it would make sense to live there.

Reply:
Got it; let's leave it here then :+1: